### ISSUE TITLE
Monster auditing for stable

### ISSUE THREAD

**Issue by I-am-Erk** [CleverRaven/Cataclysm-DDA#70215]
### Is your feature request related to a problem? Please describe.

We've had a number of gradual issues with monster types and spawn frequencies develop over the last year or so that need to be audited for stable. I am sure I've missed a few so feel free to help me out in comments.

### Solution you would like.

1. Lake and river monsters
Presently, lakes seem to all be a kind of wildlife preserve full of giant insects and angry animals swimming around the surface. The spawn rates of things like giant water striders need to be dropped precipitously - I'd suggest they be completely removed from the random lake and river spawns and instead made to spawn inside map extras that put them among clusters of reeds and other appropriate plants. Likewise, we need a looooot fewer birds and small mammals running around on the lake surface. Many of these again should be made to not spawn randomly and instead appear inside map extras depicting their habitats. In the process maybe we could add some animals that spawn only in the lower z-levels? It would be nice to see things when we dive in lakes.

2. Ferals
Ferals have started to get pretty out of hand. I understand there are some compelling options for them as midlevel enemies and I don't want to block that entirely, but as per the design doc, ferals should be quite rare. Unique ferals spawning in theme locations should be something we use extremely judiciously; think of this more like a special 'boss fight' than a common enemy for a given map special. @Karol1223 already did some good work on this but I wasn't very active at the time and didn't get much of a chance to weigh in and assess the bigger picture. Some areas of particular note:
- Ferals should not use guns. I've been saying this since they were introduced. In the past I've offered a few very limited ways we could allow it but I think, after reviewing how this argument has gone every time, we should just say "no ferals with guns" until such time as we've improved gun combat. There is no game design excuse for this. The gun being inaccurate or something you can armour against doesn't help: that just makes them a trivial enemy if you happen to know to bring the right armour, or a deadly enemy if you don't, all dependent on your luck with the RNG. We don't presently have enough gun fight code to justify this, nor is there a good lore explanation for how these low grade ferals are keeping up their ammo stock anyway. There's a reason we started with thrown rocks.
- We should rewrite any descriptions like "It looks like this feral was ___ before the cataclysm", with a few exceptions like soldiers (even then, the phrasing could be better).  I am particularly looking at mon_feral_human_archaeologist and all the mon_feral_fancy and _armoured variants. I would assume a "fancy dress" feral is someone who looks like they randomly threw on a number of mismatched fine clothes, not someone who looks like they were at a ballroom before suddenly going feral. These people have presumably been killing and eating other folks for several weeks, and still going through some of the motions of real life but in a terrible parody. As an example, our description for the archaeologist might be "Rabid and bloodshot, this person wears scraps of ill-assembled and mismatched field work clothing, and wields a bloodstained shovel.  Various trinkets and samples, including dried human body parts, dangle from their belt.", This leaves the actual nature of the feral open to interpretation. Were they part of an archaeology crew? Did they happen here recently? Don't know, doesn't matter, supposed to be up to the reader to decide.
- I haven't been to a mansion since the introduction of all these ferals, but I am concerned about the sheer number of mansion feral types here. It might be possible to find a handful of surviving ferals on exploration of a mansion, I don't think we need json entries for 8 unique ones, especially not so themed. And the crossbow feral should go for the same reason as the gun feral. Generally, the ferals you see in a mansion should be exactly the same ferals you see anywhere else. This is someone's house, not an SCA cosplay arena.

3. Slime replication is still wildly out of hand. At the very least, we should probably stop slimes from replicating if there's more than a set value of slimes in the reality bubble, with the max number defined in global_settings.json

### Additional context
As I said above, please let me know if you're aware of any other spawns that need to be reviewed. Graboids and mutant worms were a big problem for a while, for example, and I don't know if that's been repaired.

**Comment by fairyarmadillo** [comment]
Gun combat has actually improved significantly. Pistols in particular are not too deadly, even for poorly prepared characters. The feral security guard only appears in sealed labs, can't see in the dark, and does at max around 30 damage. A nasty surprise, but that's a far cry from the old feral militia that used to pop you with AR-15s in the middle of town.

**Comment by CoroNaut** [comment]
The water striders and birds in the lakes are a bit much, the player would ignore them 99.9% of the time anyways. Not to mention they block auto-movement quite a bit. It would be nice to see a kind of water delta for these creatures to inhabit. It could be a neat approach for connecting rivers to the ocean as well. The water delta itself could be even more swampy than the swamp biome.

For putting more creatures inside the lower levels of lakes, I'm slightly against it personally. Fishing might be a decent way to get food for some players, but most would probably just want to travel instead. More fish would cause a performance inconvenience. Perhaps there could be schools of fish that operate on horde logic to remedy this though. Having a pack of 10-20 fish going around is more realistic and would be a great boon to players that do fish while keeping performance better on average.

Ferals are a great addition. I don't think they need any nerf or fewer spawns. Over the course of playing, I've been shot once in the last 15-20ish times I've encountered a feral with a gun, which honestly seems reasonable. Of course they are going to be inaccurate though, they are zombies. It's like they just wave the gun around in front of themselves and stumble into hitting the trigger. This is in line with grabbing zombies that hold the player, or shocker zombies that launch a bolt of electricity, they have a vague idea of what they are doing. shocker zombies _know_ a living human when they see one and prioritize their death. Similar with ferals, they have/had the knowledge to use a gun/weapon and have the idea that the human needs to die.

**Comment by I-am-Erk** [comment]
I still have huge misgivings about a security guard with a gun at our present state. A shot with a 9mm is very possibly instantly fatal; if our game doesn't model that currently, it's actually a bug and will eventually change, so them being bugged to not be a game of russian roulette isn't an appropriate fix.

The question that needs to be asked is, "what is it about them that makes them interesting". If it's that they make a loud noise that summons other enemies, let's give them a different way to do that. If it's that they have a potential to deal significant damage at range, we can give them a way to do that as well. However, having a gun that mostly can't hit you but might kill you immediately if it does is not a great solution. It can work for turrets (even though presently they still don't work as they should) because a turret should be programmed to issue a warning before it opens fire, they're meant to be a deterrent as much as a weapon: they should have counterplay. However, a feral with a gun really should just shoot you, and if they're smart enough to remember how to scavenge ammo and properly reload it, they're also smart enough to remember how to properly aim.

**Comment by CoroNaut** [comment]
I'd definitely be against a feral being able to scavenge ammo and reload. If they currently do, that could probably be nerfed. My assumption is that they died with the gun in their hands and just naturally grip it since the gun was so imperative to have for the _human_ that died. That's similar to survivor zombies with spears. That in itself makes them interesting, and you already know they have a gun and _maybe_ ammo. That's great boon for an survivor in a pinch. The loud noise if they fire is just in line with screecher zombies.

For your argument about a single shot being fatal if you are already pretty beaten up: that could just be how it goes. A 9mm shot does 26 with 0 armor pen, Sure its 26% damage to an average survivor's body part (if unarmored), but that's what guns are made to do. The feral is still horribly inaccurate, and could even shoot zombies if there's one between you and the feral. A player could set up that situation too. The survivor would give a wide birth to ensure they handle the situation properly.

The shocker zombies launch a bolt of electricity at you which slow you down, damage _all_ body parts regardless of armor (unless very late game), and if other zombies surround you while you're stunned, you are pretty much dead immediately. I would _much_ rather a gun-wielding feral and 10 zombie friends than a shocker with 10 friends.

If ferals do get a nerf, I would just suggest reducing their possible ammo a little bit or make them spawn a little bit later in the play through, but I think they are honestly fine.

**Comment by CoroNaut** [comment]
As for a suggestion on a monster review, #69673. They are pretty imbalanced under certain conditions and have **heavy** variability in their danger to the player.

**Comment by I-am-Erk** [comment]
> I'd definitely be against a feral being able to scavenge ammo and reload.

That's the problem though. If they have a gun that fires, they must be doing this. They haven't been feral for a few minutes, they've been feral for days to weeks, using that weapon.

> For your argument about a single shot being fatal if you are already pretty beaten up: that could just be how it goes.

No, my argument is being shot with a 9mm can instantly kill you from full health, even with a fair bit of armour. I'm not talking about the current game balance of HP; the fact that you can be sure to tank a couple shots from a 9mm to the face in the current game if your health is high enough is a *bug*, and we can't rely on that as an element of their balance. Shockers do probably need some balance checks, but in comparison, shocker bolts are never going to be adjusted within the normal course of the game to the point where they're very often immediately disabling or fatal.

This entire conversation is something I've had a *ton of times* now and is why I think we need to move from "under certain conditions" to "no, it's not worth the hassle" for feral guns for the time being.

**Comment by CoroNaut** [comment]
If it were the case that a 9mm shot can instantly kill you, then a feral having a 5-10% chance to one-tap the player would indeed not be fun. I would say the same for NPCs, any turret, TALON UGV, or even grenade manhacks, which are much more deadly than a feral with 9mm. My comments are specifically from the viewpoint that weapons work as they do currently. Refactoring weapons to be that much more deadly would definitely be something to talk about.

**Comment by fairyarmadillo** [comment]
> I still have huge misgivings about a security guard with a gun at our present state. A shot with a 9mm is very possibly instantly fatal; if our game doesn't model that currently, it's actually a bug and will eventually change, so them being bugged to not be a game of russian roulette isn't an appropriate fix.
> 
> The question that needs to be asked is, "what is it about them that makes them interesting". If it's that they make a loud noise that summons other enemies, let's give them a different way to do that. If it's that they have a potential to deal significant damage at modest range, we can give them a way to do that as well. However, having a gun that mostly can't hit you but might kill you immediately if it does is not a great solution. It can work for turrets (even though presently they still don't work as they should) because a turret should be programmed to issue a warning before it opens fire, they're meant to be a deterrent as much as a weapon: they should have counterplay. However, a feral with a gun really should just shoot you, and if they're smart enough to remember how to scavenge ammo and properly reload it, they're also smart enough to remember how to properly aim.

The thing that makes them interesting is that they have potential to deal relatively enormous damage but are highly manageable thanks to their lack of night vision and scent. They demand that the player approach the area carefully and treat them with respect, which is waaaay better than the hojillion bullet sponge/hulk/acid spam enemies we have. They reward, but do not require armor, and they incentivize the player to make use of cover and stealth, which are two great systems that don't matter as much against the endless melee hordes.

Shockers are just really lame enemies and I think they should either be removed entirely or relegated to portal storms. They either completely wreck you, or they are completely shut down by a single piece of equipment. Everything you're worried about with gun enemies is actually true of shockers.

9mm is about as fatal in a feral's hands as a human's, which is to say it will kill an unarmored guy in 3 hits, or one, maybe two, if you tag them and let them bleed out. People get shot with pistols and survive all the time, so I don't see a problem here. If you are inside of a sealed XEDRA facility with no armor, you have no one to blame but yourself.

Ferals are also not particularly inaccurate compared to the player. They have decent skills and make use of them pretty well. It's inaccurate to say there's no counterplay.
1) Don't go in labs. Those are the only place they appear. This also explains why they have ammo - they used to work in the lab and now that the undead have taken over, they don't have anything to shoot at until the player arrives or an experimental mutant escapes.
2) Wear armor. This does not trivialize them, it just makes their shots more survivable. Again, they're a miniboss in a sealed location, I don't think 10 or 20 ballistic armor is a big ask.
3) Utilize the lighting and night vision systems to see them before they can see you, and either rush to melee or shoot first. Tasers also stop them from shooting. This is how you would deal with a gun toting enemy irl.

The lighting system in particular is underutilized as we have made a bad habit of training our players to mindlessly faceroll hordes. You can peek-toss a lit flashlight or flare down a hallway from a shadowed location and shoot the lit up enemies while remaining safe from ranged attacks.
4) Traps exist and are underutilized because the game is often too mindlessly easy to require them. Why would you deploy nail boards to deal with a single enemy when you can just walk up and punch them? 
5) Guns are now pretty freely available. You wind up with huge piles of them very early in the game just fighting random zombies. Any rifle will outrange and outdamage a beretta.
6) Just go ahead and take the L. We have given the player extra lives in the form of NPCs, why not use them?

As someone who plays this game for a living, I feel pretty confident arguing for the inclusion of gun ferals and robots, at least infrequently and only in places like labs and military locations. We don't seem to have a problem giving hostile NPCs guns despite their health system making them much harder to kill, what's so special about ferals?

**Comment by fairyarmadillo** [comment]
>shocker bolts are never going to be adjusted within the normal course of the game to the point where they're very often immediately disabling or fatal.

Confused by this statement. They are very often immediately disabling or fatal. I tried to nerf them by reducing their multihit ability, but getting tagged once without protection is often a death sentence, and if you have protection it's nothing.

**Comment by CoroNaut** [comment]
While shocker zombies and ferals with guns are difficult for the player to deal with, both give quite a bit to the game. I'd hate to see the ranged enemies be killed off one by one. Seeing a ranged enemy is a big hint that you need to be prepared to go into that area. Without them, I would feel that going into town would pose so much less challenge. A balance rework of enemies in general might be in order if that's the way things are going.

**Comment by I-am-Erk** [comment]
> Confused by this statement. They are very frequenty immediately disabling or fatal. I tried to nerf them by reducing their multihit ability, but getting tagged once without protection is still often a death sentence, and if you have protection it's nothing.

The point of the statement is that if we do anything to shockers, it'll be a nerf; on the other hand, any changes to gun damage are likely to be a massively dangerous buff to ferals.

I'll address the longer post in a bit but:

> I had an idea to make shockers be something created only by a specific enemy that upgrades nearby zeds into shockers, similar to how zombie masters work.

A blob singularity enemy that created shockers would be a cool addition. I wouldn't really advocate removing the shocker line altogether but I'd be fine with arguments about restricting them to close range, the lightning bolts are pretty nonsensical.

**Comment by harakka** [comment]
> I'd definitely be against a feral being able to scavenge ammo and reload. If they currently do, that could probably be nerfed. My assumption is that they died with the gun in their hands and just naturally grip it since the gun was so imperative to have for the human that died.

Ferals aren't dead. They are live humans with various degree of human faculties left.

**Comment by fairyarmadillo** [comment]
Also this is a tangent but it would also be pretty easy to make ferals pick up ammo and use it if they happened across it, even if it was stored inside a downed corpse or a piece of furniture. It'd essentially just be EAT_FOOD with one extra range and it'd fill their ammo count instead of their stomachs. But they'd be limited to a single caliber and wouldn't have any AI to go seek the stuff out, at least not until somebody codes that.

**Comment by CoroNaut** [comment]
As another monster review suggestion, "slimes". I've found that they are awesome in theory, but terrible for performance when they spread. That's not even mentioning that people play this game on tablets, ipads, etc. Then there's the fact that when they do multiply to extreme amounts, they only serve to massively increase your melee stats. My current save file slime kills: 
3450 small slimes
1153 slimes, 
157 big slimes
For reference, my "zombie" kills is 1463, and the many low-mid-tier zombies are between 50-150, the mid-late game zombies being 10-50ish.

I would like to see a nerf to the slimes' multiplication at the very least, and a slight nerf to the skill gain from killing them as a bonus.

**Comment by I-am-Erk** [comment]
The issue there, and with ferals using guns overall, is that if they're smart enough to do that, they're also smart enough to use the weapon correctly; if they are, they should be able to kill players pretty easily. Yes, you *can* survive a 9mm hit, but as I said, it's entirely possible for it to instantly kill you.

I do have less issue with ferals spawning with guns in labs, and back in the day I was cautiously willing to accept it as a possibility - with a lot of caveats - but what I see having happened over the last two years is just a lot of confusion over what a feral should be able to do, resulting in a lot of extra backtracking and fixing over the last several months. I think we need to lay out a very clear set of boundaries for what they can and can't do, and revisit it later, to avoid all this extra work and very tiring repetition of debate.

It's easy enough to imagine other ways to get similarly fun gameplay with a lab feral, not the least of which would just be to give them something deadlier than a rock to throw.

Another option, if we want to go the true miniboss route, is to make them a monster_talker entity or a hostile NPC, and allow them to start a brief conversation with the player if they spot you; they'd then be unique and isolate to particular locations. This allows us to clearly delineate this one, "smarter than usual" feral who remembers their training and has a weapon from the dudes running around with fire axes and lead pipes.

**Comment by fairyarmadillo** [comment]
I have a boss monster in the works who is locked in his own room in the bottom of a globally unique lab, and has a p90. He taunts you over the intercom as you run around the place. He is pretty cool, but the fact that you can't be surprised by him takes away a lot of the fun that organically happens encountering actually dangerous ranged enemies in an area where you can play cat and mouse while also dealing with zombies.

IDK, gunfights are cool. There are a lot of games about them because it's fun and exciting.

**Comment by ADekema** [comment]
I agree that ferals shouldn´t be using guns at least not initially. Any feral capable of operating a gun directly after the cataclysm should probably just be a hostile NPC. The only exeption should be ferals sealed in a bunker, shelter or lab where they have nothing to shot add and wouldn´t starve for a few months.

I do think it would be cool to have gun wielding ferals show up later in the game. Currently ferals just evolve into normal zombies (them dying off from violence, exposure, disease etc.). It would be more fun if some of the smarter but not humanlevel ferals survived by showing a limited capacity to adopt and learn. This wouldn´t be them regaining all or most of thier memories and intelligense as they would just turn into a NPC at that point. But just enough to figure out how to do basic survival and fight better. This is where you could get your gun wielding feral from that shoots in your general direction while sprinting at you. You can also get your armored ferals from here as they put on rudamentary and scavenged (perhaps from a survivor) bits of armour. Could also have crossbow and spear shooting ferals here.

**Comment by esotericist** [comment]
okay, i think there's some cross-communication and conflation issues going on here, so i'm going to try to address some points:

*  why is it okay for npcs to have guns but not ferals, when npcs are harder to kill?
   *  npcs might ought not be necessarily substantially harder to kill than ferals beyond armor; the thing erk has repeatedly said above about 'even a pistol should be potentially one-shot-deadly to a player' equally applies to NPCs. that's an artifact of how we handle things, and in all likelihood once wounds happen (and yes i know it's not soon, but it's still on the horizon) that will shift.
   * NPCs should in most cases be possible to talk to, reason with, or scare off. right now bandits are unwilling to communicate, but they shouldn't be our primary NPC encounters, and even bandits, right now, are possible to scare off. they don't fight mindlessly, and when presented with certain kinds of threats they will retreat. they should get pickier as we continue to enhance these systems.
     * also bandits should at least ATTEMPT to demand stuff from a player before killing them, as a rule. they don't want to get shot either.
*  turrets, ugvs, hacks, etc
    *  turrets and ugvs shouldn't be lethal surprises. turrets outside of very specific niche cases should always provide warnings (and frequently be out of ammo anyway), ugvs shouldn't be as prevalent and autonomous as they are, and again outside of very specific niche cases shouldn't be unexpectedly lethal. any places where this is still the case should be seen as a legacy of old content to vastly outdated standards, we've been trying to push away from this for years
    * grenade hacks and the like should honestly generally not exist without survivor involvement. the dispatch definitely shouldn't be found in normally generated locations because it doesn't make any sense as a local-to-this-earth encounter. that's just not a thing that makes sense on our earth, and when we had the discussions years ago to transition the zombie grenadiers to the robots, i was under the impression at the time that the plan was for them to be robots that fell through a portal or something. they should be deeply reconsidered as a concept.
      * as an aside, manhacks should not exist period. they're a ridiculous concept, and it's very clear they're an import from a game that has very different design goals than we do.
*  shockers, et al
    *  the degree to which shockers can suddenly and without meaningful consequence be lethal has been a concern for some time, but figuring out exactly what to do about it hasn't been easy. assume they're going to be made more managable somehow, don't use them as a reference point for how other things should be.
*  gunfights are cool and games have gunfights because they're fun
    *  we're not designing things with gamist design goals. this is not news. any argument towards that end is not going to get traction. guns are (or should be) deadly and should be treated with caution on enemies BECAUSE they're deadly.

**Comment by kevingranade** [comment]
From both a game balance point of view and a lore point of view ferals with guns don't make sense. I don't know if I went crazy for a while and merged them,  someone else did it what,  but giving ferals guns was a mistake.

**Comment by I-am-Erk** [comment]
It was autumn 2021, things were still pretty wild.

**Comment by PatrikLundell** [comment]
I very much doubt gun toting ferals use the same rules as players. They snap shot you from the maximum range of the gun and usually hit I (have had cases where they missed, so it's no longer 100%). You can't use darkness as labs are full of ceiling lights, and the number of places where you can out range them with a rifle are few and far between (and you rarely hit at those ranges anyway, assuming you have time for a full aim). After hitting them a ranged attack you have to run away to try to set up a new attack (as you won't one shot kill them), and you have a very small chance of pulling that off without them getting at least one shot at you due to the lack of possibilities to cut line of sights for the whole duration of the retreat. Line of sight for two turns means shot fired. Somehow engineering a melee fight is the only way to deal with them (traps would work if you had time to flee, collect traps from a stash (e.g. in you vehicle), set them up without getting ambushed (implying being a good distance away from all threats), and then somehow manage to lure the feral into them without getting shot in the process, which is unlikely to be successful).
Freely roaming UGVs in labs is about the only thing worse than gun ferals (stationary guns can be avoided by ducking back the way you came, and there are ways to deal with them if you think you have to get past them on account of them being stationary).

As mention, lightning casting monsters are often either a death sentence or the lightning part can be completely ignored when equipped with counter measures.

**Comment by fairyarmadillo** [comment]
re: shooters, That hasn't been my experience. Labs are generally not fully lit and you can shoot or smash out most lights in the game. It is a problem that they don't stop to aim. They have a long cooldown time to compensate, but ideally a shooty monster would have to stop and aim like the player, giving you a chance to break line of sight. If we could simulate that, and maybe add a message and an icon applied to the player (like how move states apply icons in most tilesets) when they can see that they're being aimed at, that might improve ranged combat in general.

**Comment by Procyonae** [comment]
I don't have time to read all the comments rn but unless I missed a change private resorts are in the same boat as mansions if noone else brought that up

**Comment by TheSaddestGoomba** [comment]
I'll second what fairyarmadillo is saying about facing feral security guards. I've played a few TCL challenge starts recently and I found them to be welcome variety. They only killed me once (as most new threats do). The inconsistent patches of light, furniture, and TCL's complex layout gave me plenty of options to deal with them but still feel pressured.

I think starting scenarios are the one place for them to really shine (same for some other feral types). Assuming you do the challenge start it is essentially minute 1 of the cataclysm. These guards would still have a mag loaded in and wouldn't have had enough targets before you to squeeze out the last few shots. I feel the only changes they need in the TCL scenario is limited ammo (no reloading) and some reduced accuracy. I think it's fitting to have gun-toting ferals upgrade into gunless ferals (or just zombies) very quickly, within days of game start. 

I don't mean to complicate the discussion but bring this up as a corollary. I have put some thought into reimplementing the 'feral diver' as a mob unique to a new challenge start inside a damaged freshwater research station. You wouldn't find them any other way, and they would be a day 1 only spawn and upgrade into new 'diver zombies' within a day (presumably finding some creative way to kill themselves). The previous concerns for them were around using their equipment and I think they simply shouldn't. Ideally they'd drown as quick as any feral. But for the brief time they're around, they provide an interesting mob tied to their surroundings and player's origin, with unique loot drops (clean diving gear) that could be key to a safe escape.

**Comment by Procyonae** [comment]
Specifically for the TCL scenario we could have some kind of "spooked security guard" like the civ monsters that does the same thing but only spawns for the first day(s) and attacks regular zeds too (but doesn't spawn near any)

**Comment by CoroNaut** [comment]
As the game is currently, ferals with a gun and limited ammo pose a decent threat to the player, as do many other unique monsters like screechers, shockers, burners, smokers, bloated zombies, etc. The balance comes with how the players handle these zombies. If the player runs head-first into a feral with a gun, yes they are going to have a hard time. A different person who recognizes that a gun pointed at them is dangerous will have a pretty easy time.

I can definitely see a half-dead feral tightly holding a gun and having the faculty to at least pull the trigger, if not just stumbling into shooting it. The lore point of view here is what exactly? A zombie hollow zombie can properly make reach attack with its arm because why? A zombie wrestler knows how to kick, punch, grab, and drag you away? What is stopping a feral from holding a gun and maybe moving their pointer finger towards themselves to fire it? What exactly is the line we are drawing here?

**Comment by PatrikLundell** [comment]
The only "safe" way to deal with gun ferals is to run as soon as you see them and hope you won't be shot in the back (and, of course, never return). Rushing a gun feral is safer than running away in many cases because it will have time to fire fewer shots. Sure, you can try to duck into a room and hope to melee it as soon as it shows up, but half of the time it has time to fire a shot at you anyway (and you can't fire at it because of the asymmetric firing paths). All of the zombies mentioned can be safely retreated from (assuming you maintain an escape path) unless you turn a corner and find yourself fact to face with them.

**Comment by OldFriendEr** [comment]
having spooked security or military  employees with limited ammo that attack everyone except their colleagues at the start of the game would liven up the world. Hearing shots in the background and then as time goes on, the surroundings becomes quiet as everyone around dies that is not dead yet.

**Comment by CoroNaut** [comment]
Sure you can get shot in the back, that's your punishment from getting too close and deciding you wanted to retreat. This can be said for pretty much any zombie that you feel you can't beat in the moment. Same for rushing a unique zombie, killing a single zombie before it's friends can close the distance is also a very viable strategy. Take acidic enemies that spew acid, or corrosive zombies that shoot at you from practically the same average distance. You would still decide "ok this is too tough" and run, or rush in. Does that mean we should remove these enemies as well?

**Comment by PatrikLundell** [comment]
Acid attacks are far less lethal than gun shots. Fire, however is worse, but the fire zombies have been scaled back for that very reason.

"Punishment for getting too close" is a silly argument, unless you mean "too close" is to enter any area where your meta knowledge of the game tells you these can be found, as you tend to encounter them when they or you turn a corner. In the former case you can duck back and run, but not in the latter, as you're unlikely to be adjacent to a corner to duck behind.

**Comment by CoroNaut** [comment]
I would definitely say that the acid attacks from corrosive, caustic and bilious zomibes are much more lethal than a feral with a gun.
This particular attack from caustic zombie does 29 damage and puts the character is severe pain, not to mention "your bleeding is beyond staunching barehanded":
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/24307650/789968f9-2afa-4455-a821-60712ef42817)
A feral prepper doing 15 damage (from when it can hit you, because its accuracy is bunk), causing minor bleeding and minimal pain:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/24307650/cd637474-042d-4d7a-abc6-9b27505008c0)

**Comment by fairyarmadillo** [comment]
The preppers should definitely go. Security guards only live in endgame dungeons and exist to make those places appropriately challenging.

**Comment by CoroNaut** [comment]
It's even less for the feral security gaurd (when they hit):
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/24307650/1dbb7d5d-ca12-49fb-91df-56adc5d38f18)

**Comment by I-am-Erk** [comment]
we are back to retreading the first few posts of this discussion again.

**Comment by CoroNaut** [comment]
Well, there seems to be a nice consensus on your solutions 1 and 3. It seems fine to go back over points that were missed or not fully thought out for ferals. Personally, I'm not really seeing where we are drawing the line at ferals with guns. They seem well within the bounds of the wacky and other much more dangerous and more 'unbalanced' enemies. It's also very hard to change or remove an enemy for _balance_ if the players skill is most of what determines the outcome of any particular interaction. Right now, enemies just evolve at a fixed rate, making it scale to the player would be the best way I can think of to balance everything.

The arguments saying that guns in feral's hands makes them trivial or deadly is basically how NPC's are. The RNG aspect of getting shot is present in all situations with all guns (unless you have high gun skill and can hit every shot). If the argument is that they _should_ be smart enough to reload if they have a gun in the first place, we could just say that isn't the case and make it so. Not to mention everything in [fairyarmadillo's comment](https://github.com/CleverRaven/Cataclysm-DDA/issues/70215#issuecomment-1856576850) about how they are an interesting enemy to face.

I did a small test, letting a "feral security gaurd" shoot me 6 times. 1 Rleg, 1 Rarm, 1 Larm, 3 torso. My character isn't exactly walking away from that easily, but this can be remedied by giving them 2-3 bullets max. We can easily assume that the living version of this guard used most of them, if not all. The argument that a feral will kill a survivor easily is a bit out there, but if it is too powerful for the evolution, then make it spawn later. It just seems like a waste to remove a unique enemy when we have the ability to make it a better enemy. 

There's also the comments about changing the lethality of weapons as a whole. We could definitely put that into another issue for balance even if it isn't worked on immediately. At the very least, it would give us some ideas to think about for the future.

**Comment by I-am-Erk** [comment]
Ultimately, while I love to have consensus, it's not really what we're looking for here. A few senior devs have explained why the present setup isn't going to remain, and the value of feedback isn't in trying to decide if we're right or not, it's just useful in trying to work out what things are worth it to be preserved. That's why I'm trying to say it's not worth continuing to circle around this topic and repeat the same arguments.

**Comment by fairyarmadillo** [comment]
It's not about winning or losing, it's about making the best game possible. The fact that after these enemies had been part of the game _for 3 years_ and then out of nowhere a dev suddenly noticed them and unilaterally decided to remove them isn't really gonna sit right, especially when you're specifically going after the well-balanced enemy and not things like mi-go scouts or talon UGVs, which shouldn't be removed either - but they are worse than the feral security guard.

There are more guns than people in this country, and now almost all of the people are dead while the guns are still around. There is no plot justification for preventing people and things from using them on the player. The current implementation of those guns is not perfect on the enemy side, but it is still mechanically satisfying and challenging. There have been some ways outlined to improve it.

Overall the game tends to suffer from enemy group and tactic homogenity, overreliance on swarms of relatively inconsequential melee enemies, and generally being too easy once you have the slightest idea what you're doing. The first issue actually gets worse as the game goes on. One of the things that makes labs exciting for players is that they present a variety of enemies that aren't the same ten zombies, and unlike mi-go etc. there's a tangible reward for doing so. It's a dungeon, and it makes in-character and game design sense that it would be especially dangerous there.

Instant kills are basically always bad game design unless the player has done something to earn it, like walking into a portal or crashing a car at high speed. Telegraphing that instant kill with a popup that more or less instructs you how to escape the instant death is not exactly better because it immediately trivializes it, to the point you may as well not even include it. It would turn what I consider one of the most exciting kinds of enemies (that being gun enemies in general) into something you are basically never threatened by. We do not need to hold the player's hand when they're crawling around in labs.

The frustrating part is really just that this issue seems to have been decided before the issue was even opened, leaving me unsure why it's an issue at all.

**Comment by I-am-Erk** [comment]
> this issue seems to have been decided before the issue was even opened. If it's not open for debate I'm not really sure why it's an issue at all.

I mean, at no point did I ask if people agreed with my initial consensus. It's an issue because it needs to be done for stable and so that people can suggest other areas that need work. However, it obviously is open for debate since I've been hearing out your opinions for a full day and shifted my initial opinion. I didn't come around to agreeing with you in full, that doesn't mean your thoughts have been ignored.

**Comment by Maleclypse** [comment]
The mansion ferals should spawn on a specific mansion start only.  If they don't work that way currently they could easily in the future. I say this because there is a specific challenge scenario they exist for and originally they only spawned in that mansion.

**Comment by I-am-Erk** [comment]
I don't like them much either way, but if they're part of a unique scenario and meant to be a bit silly I could give them more leeway.

**Comment by PGR-14** [comment]
> I agree that ferals shouldn´t be using guns at least not initially. Any feral capable of operating a gun directly after the cataclysm should probably just be a hostile NPC. The only exception should be ferals sealed in a bunker, shelter or lab where they have nothing to shot add and wouldn´t starve for a few months.
> I do think it would be cool to have gun wielding ferals show up later in the game. Currently ferals just evolve into normal zombies (them dying off from violence, exposure, disease etc.). It would be more fun if some of the smarter but not human level ferals survived by showing a limited capacity to adopt and learn. This wouldn´t be them regaining all or most of their memories and intelligence as they would just turn into a NPC at that point. But just enough to figure out how to do basic survival and fight better. This is where you could get your gun wielding feral from that shoots in your general direction while sprinting at you. You can also get your armored ferals from here as they put on rudimentary and scavenged (perhaps from a survivor) bits of armor. Could also have crossbow and spear shooting ferals here.

The ferals that spawn w/ guns should probably be either very inaccurate (They've got shaky hands/eyes, little to no trigger discipline, etc.) or their guns should be heavily damaged

The ferals becoming smarter is rooted in the lore (I think; design doc could've changed), & for the evolution thing, you could do some cool shit w/ them adapting to their environment;
For example, you could have ferals living in the city find weapons & armor (Evolving to "smart" ferals), or in the forest/wilderness getting survival skills & making a bow
For stuff like military bases, the feral soldiers there could find some weapons & gear, giving them a (VERY beat-up) rifle & tactical gear. (Another thing, what about Spec-Ops ferals? They'd most likely survive the zombies, but w/ the PTSD & mental trauma, they'd be much more prone to becoming feral)

**Comment by CoroNaut** [comment]
Making ferals use NPC code to actively seek out ammunition would definitely be _a lot_ of work and I can't see it being done. I love the idea of ferals being adapted to where they spawn though, and that is much easier.

Talking about feral soldiers and spec-ops ferals reminded me of the "Puppet soldiers" from metal gear solid 5. Of course, they could be a final-level feral soldier zombie. I would love to go against 5 of these things while they make a mad dash for me.
[Metal gear solid wiki for "Puppet soldiers"](https://metalgear.fandom.com/wiki/Puppet_soldiers)
<img src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fget.wallhere.com%2Fphoto%2Fvideo-games-soldier-Metal-Gear-Solid-Metal-Gear-Toy-machine-Metal-Gear-Solid-V-The-Phantom-Pain-screenshot-109079.png&f=1&nofb=1&ipt=4e8ad17f6bca1a1dd78bf3a7d8d90d0fdae74a0c710e5ce65f06af1266f9d068&ipo=images" width="360">

**Comment by fairyarmadillo** [comment]
> Making ferals use NPC code to actively seek out ammunition would definitely be _a lot_ of work and I can't see it being done.

That's incorrect, and it will be trivial to do as soon as I give animals code to actively seek out food.

> The ferals that spawn w/ guns should probably be either very inaccurate (They've got shaky hands/eyes, little to no trigger discipline, etc.) or their guns should be heavily damaged

Hard disagree. If they are smart enough to work out how to use a gun, then they should be about as skilled as they would have been pre-feralization. For a random citizen that might not be much, but for a security guard or soldier it might be quite high.

> The ferals becoming smarter is rooted in the lore

I don't know where you're getting that but afaik it isn't true.

This is a zombie survival game, not a feral survival game. Having the odd enemy that can break the normal zombie rules is fantastic. Making thousands of ever-evolving ferals with a wide variety of skills and equipment is counter to the core of our game design.

**Comment by CoroNaut** [comment]
I didn't realize that allowing making ferals seek ammo would be trivial as you say. It seems there isn't a consensus on the level of intelligence for ferals. We have a wide range of possibilities for how smart they should be. 

1. Ferals throw rocks/items to be a slightly larger threat than normal zombies
2. The feral doesn't _try_ to fire their weapon, it just happens because the feral is holding the weapon and stumble into firing
3. The feral knows what a gun does and actively _attempts_ to aim and fire at its target
4. The feral understands that the weapon uses ammunition, but not where to find it. It might just come across some and be lucky, then pick it up and reload.
5. The feral has almost human-level intelligence, and can actively seek out ammunition to maintain itself as a major threat.

As for how smart they are:
1. Evolution could upgrade them over time. This idea would be definitely be cool
2. Ferals could spawn in one of the above categories and just not evolve. Perhaps the lore could be like: "Some humans held onto their mental state for as long as they could through sheer will, but yet, all have eventually succumbed to the virus"

I would also be against having thousands of them, but perhaps 1 in every 2~5 overmap tiles would make a nice change of pace. Having them use NPC code would definitely be iffy, Having them stay as zombies that are unable to move beyond the reality bubble seems like a good middle ground. I would think that humans turned feral in random towns or the forest would have basic knowledge, and those in military bases would be more deadly.

**Comment by PGR-14** [comment]
> I don't know where you're getting that but afaik it isn't true.

Unless the design doc is incorrect it says it in the Lore - Setting location

Cognition The blob has a potent, albeit often subtle, effect on human cognition.
In about one quarter of the population, changes are basically unnoticeable.
For about half the population, the blob causes an increase in risk-taking behaviour, ranging from mild unsafe practices all the way to people attacking hordes of zombies with a stick. This effect may be permanent or may be temporary, and resolves at varying speed by person. Some may be strongly enough affected to be almost indistinguishable from ferals, but not aligned with zombies.
The remaining quarter see a heavy increase in aggressive, violent behaviour, often completely out of character. The worst of these, about 1/20 of the population, become “ferals”, a form of living zombie. These ferals are not seen as hostile to zombies, and mutate naturally with time just as zombies do. Ferals still maintain a degree of human intelligence, depending on how severely affected they are… Some can remember tool use, for example. Ferals led to a great deal of confusion over whether the reports of the dead rising were true, as feral behavior is very similar to that of zombies. From an out-of-universe perspective, ferals are similar to “rage virus” zombies as seen in 28 days later, but adapted to our lore, and as such are considered zombies internally.
“Blob psychosis” is an out-of-game term for the increased violence experienced by this quarter of the population. Nobody in game calls it that. Strictly speaking it’s not a psychosis, but that term seems to have stuck in our game discussions.

COGNITION AFTER THE CATACLYSM
Over time, most affected people acclimate and return to normal. Some remain somewhat less risk-averse than before (eg players). Those that tended towards high violence also drift back towards normal cognition slowly, although their actions and the memory of them likely leave them changed permanently.
True ferals are permanently altered. Like all living humans, their numbers were severely thinned during the Cataclysm; the fact that they weren’t targeted by zombies is balanced by the fact that they didn’t make much effort to avoid confrontation with police and military forces.
Ferals Now The majority of feral humans are difficult to distinguish from zombies. They can grasp and use simple tools like clubs and melee weapons, and will avoid dangerous obstacles, open doors, and other very simple actions largely governed by motor memory, but that’s about it. These comprise around 1% of the zombie faction.
About a third of ferals remember more complex things. These ferals might use complex tools properly, and would understand for example how to use a gun, activate a Hack, or put on a kevlar vest. They lack forward thinking and executive reasoning, and would be unlikely to engage in complex tactics (they might know how to arm c4, but would not think to put it onto your base wall to blast a way in) but can nevertheless be extremely dangerous. These comprise around 0.3% of the zombie faction.
A very dangerous tenth or so of ferals maintain most of their human intelligence. They can use any tools they would have in life, can plan tactics, and can assess your weaknesses and use salvaged materiel to exploit them… Basically anything an NPC should be able to do. Likely they can talk as well, although probably it would be difficult for them to be too deceptive. These comprise about 1/1000 of the zombie faction. Over time these will probably evolve into a variant living form of zombie Masters and become another nemesis level villain for the late game. Feral Masters would be exceedingly rare, probably 1/10,000 or less, given the low odds of an intelligent feral living long enough to evolve.

**Comment by fairyarmadillo** [comment]
Not sure what part you're referring to. The thing about intelligence returning is referring to some people only temporarily going feral, prior to the start of the game, and recovering thereafter, either becoming survivors or bandits.

The thing about feral masters is talking about a creature so rare that it would probably be unique for each run, essentially a boss monster like the shadow liutenant. They're not evolving into feral commandos.

**Comment by CoroNaut** [comment]
I'm pretty certain the current form of ferals closely matches this already:

> The remaining quarter see a heavy increase in aggressive, violent behaviour, often completely out of character. The worst of these, about 1/20 of the population, become “ferals”, a form of living zombie. These ferals are not seen as hostile to zombies, and mutate naturally with time just as zombies do. 

If they mutate naturally just like regular zombies, who's to say there isn't a feral hulk, feral shocker, feral spec-ops brute? Sure their intelligence probably wouldn't increase from just mutating. If 1% of the zombie faction is ferals, it seems reasonable their intelligence could be all over the place.

> They can grasp and use simple tools like clubs and melee weapons, and will avoid dangerous obstacles, open doors, and other very simple actions largely governed by motor memory, but that’s about it. These comprise around 1% of the zombie faction.

I would personally put holding a weapon and firing it in a limited way under this category. If the feral can grasp a door handle and push/pull, they can grasp a weapon and push/pull the trigger, especially if they have motor memory. That doesn't even mention: "About a third of ferals remember more complex things. "

> For about half the population, the blob causes an increase in risk-taking behaviour, ranging from mild unsafe practices all the way to people attacking hordes of zombies with a stick. This effect may be permanent or may be temporary, and resolves at varying speed by person. Some may be strongly enough affected to be almost indistinguishable from ferals, but not aligned with zombies

If "ferals" are those who are highly aggressive and not seen as hostile to zombies, then there should be another form of zombie or half-living being who is at the very least, unpredictable in their aggression. Perhaps a "half-feral" zombie? Perhaps these "half-ferals" can be permanently this way, or have a set date when they return to being an NPC? If they aren't aligned with zombies, one could assume that most, if not all of these "half-ferals" are dead. If they attack every zombie they see because of their aggression, they won't last long, and we can probably not even bother spawning this form of "half-feral".

As for the feral masters being a boss like the shadow lieutenant, I'd agree. Perhaps make them spawn once per 180x180 overmap region instead of one unique for the entire world though. Perhaps their level of intelligence could be a major part of what kind of boss they evolve into? We could have a range of feral bosses that go essentially from a very quick, very dextrous feral brute/hulk wrestler to a spec-ops brute soldier zombie that wields an AR-15. Simply wielding a weapon/gun doesn't rely on anything other than basic intelligence, unless they just couldn't hold it. The stats the feral has or gains as a result of their evolution would be a vital part of exactly how _effective_ using that weapon/gun is.

**Comment by I-am-Erk** [comment]
I'd really appreciate if this conversation were taken to a discussion thread, it's not particularly connected to auditing monsters for stable.

**Comment by fairyarmadillo** [comment]
Does anyone know what causes slimes to multiply? I was trying to test a fix for them multiplying out of control and discovered that no matter what I tried, slimes and big slimes would never increase their numbers. Sometimes big ones will split into smaller ones, but they just reform after a while and the net number doesn't go up.

**Comment by PatrikLundell** [comment]
I thought they were supposed to be able to split upon acquiring more "HP", but otherwise they'd just split and reform in an annoying lag inducing manner.

However, slimes can "convert" monsters into slimes by splitting onto them, so I suspect that would be a source of additional raw material for slime formation (but maybe it just eliminates the monster, leaving the splinter with the same "HP" as it acquired upon splitting).

Without knowing the implementation, I'd introduce a hysteresis by disallowing slimes to split spontaneously if they have too few "HP" (which probably is in place), but also to merge if the merged slime would get too many (which would be a lower limit than the split one to at least forbid these two slimes to join up again).

**Comment by fairyarmadillo** [comment]
> However, slimes can "convert" monsters into slimes by splitting onto them, so I suspect that would be a source of additional raw material for slime formation (but maybe it just eliminates the monster, leaving the splinter with the same "HP" as it acquired upon splitting).

I was not observing this when i gave them lots of things to kill. Even if they did do it and I just wasn't seeing it for some reason, that would not lead to exponential growth like I've seen in some people's reddit posts, as it would be eliminating one monster to make one or two more, and they'd eventually run out of monsters.

My idea was to map a fairly large radius when FORMBLOB procced and count the monsters in the vicinity, then return out of the function if there were more than, say, 30. But again I couldn't reproduce the issue to test whether that solution would work.

**Comment by RenechCDDA** [comment]
Presumably we want this to remain open until it is actually done.

**Comment by Inglonias** [comment]
we really should tell the automation bot to consider issues in projects "hands off"

**Comment by I-am-Erk** [comment]
Gun ferals are resolved now. I believe lake monster spawns were fixed shortly after the issue was posted. If we've got confirmation that the silly mansion zombies are scenario specific, I think that just about has us covered - just need slimes fixed.

I *would* still like to revisit the actual role of guns in the ferals they were taken from, rather than remove them. I agree with the above discussion that they added an interesting play function we shouldn't just be discarding.

**Comment by I-am-Erk** [comment]
I'm going to call this done for 0.H stable, but I really want to focus on ferals. The current "take guns away from them" change is the least satisfying fix for this issue, I do still want to look at what made gun ferals interesting and restore it to the game.

**Comment by harakka** [comment]
Nothing has been done about #1, rivers and lakes are still full of water striders and water scorpions and all that goodness.

**Comment by ehughsbaird** [comment]
What is the status of this?

**Comment by I-am-Erk** [comment]
I can try to run through some of these and check tonight if i have time but I'll cross post to discord and see if anyone has some up to date playtest info.

**Comment by I-am-Erk** [comment]
I'm also reminded that I really want to go back to gun ferals and shocker zombies and figure out how to get their gameplay function into the game without the associated problems.
